app.js

```diff
@@ -0,0 +1,130 @@
+/*
+    *** Conventions for this program ***
+
+    *** For the POST API ***
+        DATA FORMAT in JSON is {
+                                    "data": [1,2,3,4,5......]
+                               }
+
+    *** For the PATCH API ***
+        DATA FORMAT in JSON is {
+                                    "num": 5
+                               }
+
+    *** Tested the API calls in POSTMAN
+
+
+*/
+
+const express = require('express');

```

Comment:
How did you decide on using JavaScript for the API server?  What are the advantages and disadvantages to using JavaScript for this purpose?

Reply:
Writing API server using JavaScript (Node.js) is fairly easy and using Express on top of that easily reduces production time. The only concern for this implementation is sorting 500 numbers which in worst cast has O(nlgn) time complexity. This is nothing Node.js can not handle. However, if we have to change the implementation to sort a way more than 500 numbers and if we have heavy GET requests (which sorts the number in the current implementation), the computation could bottlenect the performance of the system by blocking other incoming requests.